contracts/NFTMarketplace.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import '@openzeppelin/contracts/utils/Counters.sol';
+import './MarketItem.sol';
+
+contract NFTMarketplace {
+    MarketItem private marketItem;
+    using Counters for Counters.Counter;
+    Counters.Counter private _collectionIds;
+    uint256 public LISTING_FEE = 0.0001 ether;
+    address payable private _marketOwner;
+
+    mapping(uint256 => Collection) public collectionLedger;
+    mapping(uint256 => MarketNft) public nftLedger;
+    mapping(address => uint256) private credits;
+
+    struct Collection {
+        uint256 collectionId;
+        string seller;
+    }
+
+    struct MarketNft {
+        uint256 tokenId;
+        uint256 collectionId;
+        uint256 price;
+        bool forSale;
+    }
+
+    event CollectionCreated(uint256 indexed collectionId, string seller);
+
+    event MarketNftCreated(
+        uint256 indexed tokenId,
+        uint256 collectionId,
+        uint256 price,
+        bool forSale
+    );
+
+    constructor(address _marketItemAddress) {
+        marketItem = MarketItem(_marketItemAddress);
+        _marketOwner = payable(msg.sender);
+    }
+
+    function createCollection(string memory collection) public payable {
+        require(bytes(collection).length != 0, 'Collection cannot be empty');
+
+        _collectionIds.increment();
+        uint256 newCollectionId = _collectionIds.current();
```

Comment:
I beleive you need an array to keep collections ids, in order to be possible to itterate collections from the Front end